src/mongo/db/catalog/collection.cpp

```diff
@@ -1129,7 +1129,7 @@ class RecordStoreValidateAdaptor : public ValidateAdaptor {
 
                 if (indexEntryCount != 0) {
                     indexEntryCount--;
-                    dassert(indexEntryCount >= 0);
+                    //dassert(indexEntryCount >= 0);
```

Comment:
indexEntryCount is unsigned integer variable. If it can be negative in your mind consider using signed integer variable.